configuration/package_scripts/tedge-p11-server/preinst

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+set -e
+
+command_exists() {
+    command -V "$1" >/dev/null 2>&1
+}
+
+group_exists() {
+    name="$1"
+    if command_exists id; then
+        id -g "$name" >/dev/null 2>&1
+    elif command_exists getent; then
+        getent group "$name" >/dev/null 2>&1
+    else
+        # Fallback to plain grep, as busybox does not have getent
+        grep -q "^${name}:" /etc/group
+    fi
+}
+
+user_exists() {
+    name="$1"
+    if command_exists id; then
+        id -u "$name" >/dev/null 2>&1
+    elif command_exists getent; then
+        getent passwd "$name" >/dev/null 2>&1
+    else
+        # Fallback to plain grep, as busybox does not have getent
+        grep -q "^${name}:" /etc/passwd
+    fi
+}
+
+### Create groups
+if ! group_exists tedge; then
+    if command_exists groupadd; then
+        groupadd -g 992 --system tedge
+    elif command_exists addgroup; then
+        addgroup -g 992 -S tedge
+    else
+        echo "WARNING: Could not create group: tedge" >&2
+    fi
+fi
+
+### Create users
+# Create user tedge with no home(--no-create-home), no login(--shell) and in group tedge(--gid)
+if ! user_exists tedge; then
+    if command_exists useradd; then
+        useradd -u "999" --system --no-create-home --shell /sbin/nologin --gid tedge tedge
```

Comment:
I guess that using a well-know user and group id (aka  `992` and `999`) is to ease shared-access to the socket across containers. However, here this sounds more as magic numbers, as these are documented no where.

Reply:
Actually, after thinking about it a bit more, I'll remove the fixed assignment, and just make it a pre-requisite for users to setup before the package is installed (which can be documented in the future PR doc).

Reply:
Also, tedge-p11-server is supposed to be installed on a host, so we have to create user and group here as we do in the preinst script for tedge, but don't we also have to make sure the ids match for access control?

Reply:
Changed in 084b00ff0f6eb400fdb7a33cb4ae94a85c950c51

The usage of explicit GID and UID has been removed to avoid conflicts where the GID/UID are already assigned to different users. The downside is that the user MUST now configure a static GID/UID before using it in any container environments (but this is generally done anyway on the image building layer, so shouldn't be a big barrier).